### PR TITLE
Change jest to always use jest-junit reporter whether in CI or not

### PR DESCRIPTION
**Pull request by curtisman**
To mirror what mocha does so that local run can be more similar to CI runs

### CODE REVIEW

File: examples/apps/collaborative-textarea/jest.config.js [hunk 1]
```diff
@@ -20,7 +20,27 @@ module.exports = {
...
+			{
+				outputDirectory: "nyc",
+				outputName: "jest-junit-report.xml",
+			},
+		],
+	],
 	// While we still have transitive dependencies on 'uuid<9.0.0', force the CJS entry point:
 	// See: https://stackoverflow.com/questions/73203367/jest-syntaxerror-unexpected-token-export-with-uuid-library
 	moduleNameMapper: { "^uuid$": "uuid" },
+	reporters: [
```
Comment (tylerbutler) [inline examples/apps/collaborative-textarea/jest.config.js:36]: Is this a duplicate entry?

File: examples/data-objects/clicker/jest.config.js [hunk 2]
```diff
@@ -20,7 +20,27 @@ module.exports = {
...
+			{
+				outputDirectory: "nyc",
+				outputName: "jest-junit-report.xml",
+			},
+		],
+	],
 	// While we still have transitive dependencies on 'uuid<9.0.0', force the CJS entry point:
 	// See: https://stackoverflow.com/questions/73203367/jest-syntaxerror-unexpected-token-export-with-uuid-library
 	moduleNameMapper: { "^uuid$": "uuid" },
+	reporters: [
```
Comment (tylerbutler) [inline examples/data-objects/clicker/jest.config.js:36]: Same here.

File: examples/data-objects/todo/jest.config.js [hunk 3]
```diff
@@ -20,7 +20,27 @@ module.exports = {
...
+			{
+				outputDirectory: "nyc",
+				outputName: "jest-junit-report.xml",
+			},
+		],
+	],
 	// While we still have transitive dependencies on 'uuid<9.0.0', force the CJS entry point:
 	// See: https://stackoverflow.com/questions/73203367/jest-syntaxerror-unexpected-token-export-with-uuid-library
 	moduleNameMapper: { "^uuid$": "uuid" },
+	reporters: [
```
Comment (tylerbutler) [inline examples/data-objects/todo/jest.config.js:36]: And here.

**Review by tylerbutler — APPROVED** [review]

File: examples/data-objects/todo/jest.config.js (hunk 3 above)
Comment (curtisman) [reply examples/data-objects/todo/jest.config.js]: fixed.

File: examples/data-objects/clicker/jest.config.js (hunk 2 above)
Comment (curtisman) [reply examples/data-objects/clicker/jest.config.js]: fixed.

File: examples/apps/collaborative-textarea/jest.config.js (hunk 1 above)
Comment (curtisman) [reply examples/apps/collaborative-textarea/jest.config.js]: fixed.

**Pull request merged**